src/GameLogic/Entities/Protagonist.ts

```diff
@@ -42,6 +42,13 @@ export class Protagonist implements Entity {
 			this.x += direction.x;
 			this.y += direction.y;
 		}
+
+		if (level.tilesFloor.get(this.x, this.y) == FloorType.Water) {
+			level.entities.removeEntity(this);
+			if (this.isPlayerControlled) {
+				// @ todo figure out how to reset the level if this is the player-controlled protagonist
```

Comment:
I think the reset should be handled by the UI - the player might want to see the state upon death (and maybe we can let them continue passing the turns so that they can observe how other things play out)

Reply:
Yes, you are right. I will remove this.